GTG/core/plugins/api.py

```diff
@@ -146,21 +147,42 @@ def remove_active_selection_changed_callback(self, plugin_class):
     def add_menu_item(self, item):
         """Adds a menu entry to the menu of the Task Browser or Task Editor.
 
-        @param item: The Gtk.MenuItem that is going to be added.
+        @param item: The Gio.MenuItem that is going to be added.
         """
-        menu_box = self.__builder.get_object('menu_box')
-        menu_box.add(item)
-        menu_box.reorder_child(item, 1)
-        menu_box.show_all()
+        _, _, menu = self.__builder.get_object(
+            'editor_menu' if self.is_editor() else 'main_menu'
+        ).iterate_item_links(0).get_next()
+        menu.append_item(item)
 
     def remove_menu_item(self, item):
         """Remove a menu entry to the menu of the Task Browser or Task Editor.
 
-        @param item: The Gtk.MenuItem that is going to be removed.
+        @param item: The Gio.MenuItem that is going to be removed.
         """
-
-        menu_box = self.__builder.get_object('menu_box')
-        menu_box.remove(item)
+        # you cannot remove items by identity since there values are simply copied
+        # when adding a new one. A reliable solution is to instead find the first one
+        # with the same label as the given one.
```

Comment:
Maybe for the future, require plugins to insert some kind of UUID as an [attribute](https://lazka.github.io/pgi-docs/#Gio-2.0/classes/MenuItem.html#Gio.MenuItem.set_attribute) to work from? So then we don't have to do some label matching. (So maybe just put an `# TODO` here or so about this)

Edit: Or maybe also check action (if set)?